tests/ga/extension_actor.py

```diff
@@ -0,0 +1,465 @@
+    A collection of static methods providing some basic functionality for the ActionSet
+    class' actions.
+    """
+
+    @staticmethod
+    def SucceedAction(*args, **kwargs):
+        """
+        A nop action with the correct function signature for ActionSet actions.
+        """
+        return 0
+    
+    @staticmethod
+    def FailAction(*args, **kwargs):
+        """
+        A simple fail action with the correct function signature for ActionSet actions.
+        """
+        raise ExtensionError("FailAction called.")
+
+    
+    @staticmethod
+    def _wrap_with_write_status(cmd, name, version):
+        """
+        A function factory for an action that executes the given command after writing 
+        a simple (successful) status file in the appropriate directory. 
+        
+        Useful in particular for wrapping "enableCommand"s, as the agent expects any 
+        extension to write such a status after enabling.
+        """
+        import azurelinuxagent.common.utils.fileutil as fileutil
+        import azurelinuxagent.common.conf as conf
+        
+        filedir = "/".join([
+            conf.get_lib_dir(),
+            Formats.FormatExtensionDir(name, version),
+            "status"
+        ])
+
+        msg = """
+            [{
+                "status": {
+                    "status": "success"
+                }
+            }]
+        """
+
+        def wrapped_cmd(*args, **kwargs):
+            filename = "/".join([filedir, "{0}.status".format(wrapped_cmd.inc)])
+            fileutil.write_file(filename, msg)
+
+            # Every time enable is called, the agent looks for the next status file
+            # (by filename prefix). We need to keep track of that number ourselves
+            # to write to the correct location.
+            wrapped_cmd.inc += 1
+            
+            return cmd()
+        
+        # this is the default value for the function's static variable; because each invocation
+        # of this function generates a distinct wrapped_cmd, we shouldn't get collisions.
+        wrapped_cmd.inc = 0
+
+        return wrapped_cmd
+
+def get_extension_actor(name="OSTCExtensions.ExampleHandlerLinux", version="1.0.0", continueOnUpdateFailure=False, 
+    updateMode="UpdateWithInstall", data_fetcher_base=mockwiredata.DEFAULT_FETCHER, installAction=Actions.SucceedAction,
+    uninstallAction=Actions.SucceedAction, updateAction=Actions.SucceedAction, enableAction=Actions.SucceedAction,
+    disableAction=Actions.SucceedAction):
+    """
+    Factory method for ExtensionActor class. Note that the provided name and version needs to match a plugin listed
+    in the xml doc returned by data_fetcher_base["manifest"]; otherwise, the agent won't be able to properly download
+    this extension.
+    """
+    
+    actionSet = ActionSet(installAction, uninstallAction, updateAction, enableAction, disableAction, name, version)
+    info = ExtensionInfo(name, version, continueOnUpdateFailure, updateMode)
```

Comment:
The action set and info are basically the extension manifest + name & version.

Consider merging them into a single entity with "ExtensionManifest" somewhere in its name, that way the test code is a little closer to the prod code. If you do that, please add reportHeartbeat to have a full manifest (no need to add rebootAfterInstall since it is not implemented). Also, get_extension_actor could take an instance of this entity as parameter (right now it takes quite a few parameters).

Reply:
Unfortunately, all of the parameters of `get_extension_actor` are necessary bits of info required to create an `ActionSet` and `ExtensionInfo`. I agree that the two classes ought to be merged, but giving `get_extension_actor` such a merged object simply shifts the parameters from a factory method to an init method...

It seems better to simply leave the parameters where they are. Thoughts?